### PR TITLE
PTX-18831 Improving logic for when Telemetry should be enabled/disabled

### PR DESCRIPTION
**Pull request by nikolaypopov**
Improving logic for when Telemetry should be enabled/disabled

### CODE REVIEW

File: pkg/util/test/util.go [hunk 1]
```diff
@@ -3017,88 +3044,231 @@ func ValidateTelemetry(pxImageList map[string]string, cluster *corev1.StorageClu
...
 	}
+	return false
+}
 
-	// Update Telemetry status
-	if cluster.Spec.Monitoring != nil && cluster.Spec.Monitoring.Telemetry != nil {
-		cluster.Spec.Monitoring.Telemetry.Enabled = newCluster.Spec.Monitoring.Telemetry.Enabled
-	} else {
-		cluster.Spec.Monitoring = newCluster.Spec.Monitoring
+func getArcusTelemetryLocation(cluster *corev1.StorageCluster) string {
```
Comment (ezhang-px) [inline pkg/util/test/util.go:3233]: This function will make our default location to be external(prod) unless specified in the annotation. Hope that's the expectation here

**Review by ezhang-px — COMMENTED** [review]
some of the APIs (GetPxProxyEnvVarValue, CanAccessArcusRegisterEndpoint, ParsePxProxyURL, IsCCMGoSupported) are redefined in other packages: drivers/storage/portworx/component, drivers/storage/portworx/util with even different implementation, can we simply call them for the test purpose instead of rewriting?

**Comment by nikolaypopov** [comment]
> some of the APIs (GetPxProxyEnvVarValue, CanAccessArcusRegisterEndpoint, ParsePxProxyURL, IsCCMGoSupported) are redefined in other packages: drivers/storage/portworx/component, drivers/storage/portworx/util with even different implementation, can we simply call them for the test purpose instead of rewriting?

I tried that and it is not possible due to `import cycle not allowed`, also as per @piyush-nimbalkar we do not want to re-use much of operator functions, just in case a bug is introduced there.

File: pkg/util/test/util.go (hunk 1 above)
Comment (nikolaypopov) [reply pkg/util/test/util.go:3233]: It is expected, this is a general validation for both production and staging telemetry

**Comment by ezhang-px** [comment]
> > some of the APIs (GetPxProxyEnvVarValue, CanAccessArcusRegisterEndpoint, ParsePxProxyURL, IsCCMGoSupported) are redefined in other packages: drivers/storage/portworx/component, drivers/storage/portworx/util with even different implementation, can we simply call them for the test purpose instead of rewriting?
> 
> I tried that and it is not possible due to `import cycle not allowed`, also as per @piyush-nimbalkar we do not want to re-use much of operator functions, just in case a bug is introduced there.

Got you, thanks for the explanation. @jrivera-px also described more context to me offline about what's going yesterday. Just one catch here about the CanAccessArcusRegisterEndpoint, I think in the current release, it's already updated, so is it on purpose that we have to use the code before the https changes?
https://github.com/libopenstorage/operator/blob/px-rel-23.7.0/drivers/storage/portworx/component/telemetry.go#L1036

**Review by ezhang-px — APPROVED** [review]
LGTM

**Pull request merged**